docs/book/src/topics/control-plane-endpoint.md

```diff
@@ -0,0 +1,215 @@
+    --> F_Octavia
+    F_Octavia --> F_CP0
+    F_Octavia --> F_CP1
+    F_Octavia --> F_CP2
+
+    D_Client --> D_VIP
+    --> D_Octavia
+    D_Octavia --> D_CP0
+    D_Octavia --> D_CP1
+    D_Octavia --> D_CP2
+```
+
+### Parameters
+
+To enable the Octavia loadbalancer, `apiServerLoadBalancer.enabled` must be `true`.
+
+The following parameters are also used by the Octavia loadbalancer.
+
+- controlPlaneEndpoint
+- apiServerLoadBalancer
+  - enabled
+  - additionalPorts
+  - allowedCIDRs
+  - provider
+- disableAPIServerFloatingIP
+- apiServerFloatingIP
+- apiServerFixedIP
+- apiServerPort
+
+### Default behaviour
+
+An Octavia loadbalancer will be created. VIP will be allocated automatically from the default cluster subnet.
+
+`controlPlaneEndpoint.port` will be set to `6443`.
+
+A floating IP will be allocated from the cluster's external network and attached to the Octavia VIP. `controlPlaneEndpoint.host` will be set to the floating IP.
+
+### No floating IP
+
+If your clients can route to the cluster network there is no need for a floating IP as clients can connect directly to Octavia's VIP. To enable this, set `disableAPIServerFloatingIP` to `true`.
+
+If required, you can choose a specific IP for the VIP either by setting it in `apiServerFixedIP` or by setting `controlPlaneEndpoint.host` directly. In this case, the specified IP must be a valid IP in the default cluster subnet.
+
+### Using a pre-created floating IP
+
+To use a specific floating IP it must be created in advance. This is because creating a floating IP with a specific IP requires OpenStack admin privileges. It is not recommended to give CAPO OpenStack admin privileges.
+
+The pre-created floating IP must be specified either in `apiServerFloatingIP` or in `controlPlaneEndpoint.host`. Note that if it is not specified in `apiServerFloatingIP` it will be deleted when the cluster is deleted.
+
+### Deletion of the floating IP
+
+When using Octavia, the loadbalancer's floating IP will be deleted when the cluster is deleted unless it is specified in `apiServerFloatingIP`.
+
+If you need to keep a floating IP which was created automatically with the cluster, it is possible to set `apiServerFloatingIP` to the current floating IP after the cluster has been created.
+
+### Exposing additional control plane services running on other ports
+
+If the control plane machines are hosting services other than the kubernetes API server, these may also be exposed via the API server loadbalancer by specifying their ports in `apiServerLoadBalancer.additionalPorts`. If using managed security groups, these ports will be automatically added to the control plane security group.
+
+### Restricting access to the loadbalancer endpoint
+
+It is possible to restrict access to the control plane endpoint to certain source address ranges by specifying them in `apiServerLoadBalancer.allowedCIDRs`. When setting this list, the following additional CIDRs will be automatically added to the allowed list:
```

Comment:
Might be worth to redirect user to the configuration section of this field for further information:
https://cluster-api-openstack.sigs.k8s.io/clusteropenstack/configuration#restrict-access-to-the-api-server